scripts/build_climate_projections.py

```diff
@@ -0,0 +1,480 @@
+        Dataset of the changes in the monthly averages for each grid cell
+    month: float
+        A month value to be considered further for the morphing procedure
+    cutout_param_name: string
+        A name of a cutout parameter for which a projection is being calculated
+
+    Output
+    -------
+    cutout_xr: xarray of the projected changes for each grid cell for a processed month
+    Contains a time series for each grid cell.
+    """
+
+    k_time = cutout_xr.time.dt.month.isin(month)
+
+    # stretch calculations
+    if a & t_mean_month:
+        dt_xr = dt_xr + a * (cutout_xr[cutout_param_name][k_time, :, :] - t_mean_month)
+
+    for i in range(0, len(cutout_xr.y)):
+        for j in range(0, len(cutout_xr.x)):
+            cutout_xr[cutout_param_name][k_time, i, j] = np.add(
+                cutout_xr[cutout_param_name][k_time, i, j],
+                # spreading dt_xr which is of monthly resolution across the whole time dimension of a cutout
+                # which is normally of hourly resolution
+                np.array([dt_xr[i, j].item()] * k_time.sum().item()),
+            )
+
+    return cutout_xr
+
+
+def build_cutout_future(
+    cutout_xr,
+    cmip6_xr,
+    months,
+    year0,
+    year1,
+    years_window,
+    cutout_param_name="temperature",
+    add_stretch=False,
+    cmip6_nn_xr=False,
+    cmip6_xx_xr=False,
+):
+    """
+    Calculate the projection for the cutout data applying the morphing approach
+    for each month in the months.
+
+    Parameters
+    ----------
+    cutout_xr: xarray
+        Cutout dataset loaded as xarray
+    cmip6_xr: xarray
+        CMIP6 climate projections loaded as xarray
+    months: list
+        Months values to be used as a definition of a season to be considered
+        for building projection time series
+    year0: integer
+        The first year of an earlier period in the future
+    year1: integer
+        The first year of a later period in the future
+    years_window: integer
+        Width of the considered time period
+
+    Output
+    -------
+    cutout_xr: xarray
+        Dataset on the projected changes for each grid cell for a processed month
+        Contains a time series for each grid cell.
+    """
+    for k_month in [months]:
+        dt_k = calculate_proj_of_average(
+            cmip6_xr=cmip6_xr, month=k_month, year0=year0, year1=year1, years_window=5
+        )
+
+        a = False
+        t_mean_month = False
+
+        if add_stretch:
+            # TODO Check cmip6_nn_xr and cmip6_xx_xr are not empty
+            dt_nn_k = calculate_proj_of_average(
+                cmip6_xr=cmip6_nn_xr,
+                month=k_month,
+                year0=year0,
+                year1=year1,
+                years_window=5,
+            )
+            dt_xx_k = calculate_proj_of_average(
+                cmip6_xr=cmip6_xx_xr,
+                month=k_month,
+                year0=year0,
+                year1=year1,
+                years_window=5,
```

Comment:
may this be a parameter?

Reply:
It must be, I have forgotten to use it there. Fixed now. Thanks :)